### ISSUE TITLE
🟢 onunload 落盘 fire-and-forget 仍 await flushTMVault 访问 vault 时序不保证

### ISSUE THREAD

**Issue by miaoziguan** [miaoziguan/obsidian-chinese-plugin-market#31]
## 位置
`src/app/plugin.ts#356-357`（onunload 落盘路径）

## 根因
```ts
if (pendingSettings || pendingTranslator) {
    void (async () => {
        if (pendingSettings) await this._saveSettingsImmediate();
        if (pendingTranslator) await this._saveTranslatorDataImmediate();
    })().catch((e) => logger.warn(...));
}
```

`_saveTranslatorDataImmediate` 内部第一行 `await this.flushTMVault()`（`plugin.ts#563`）通过 `noteStorage` 访问 `this.app.vault` 写 TM 笔记。onunload 是 fire-and-forget 且 Obsidian 已开始拆卸插件，此时写 vault 笔记大概率失败（或时序不可靠），导致防抖窗口内未落盘的 TM 变更兜底**实际不生效**——与 `#343-345` 注释"防止刚编辑的词条静默丢失"的目标相悖。

## 修复建议
卸载路径只写 `data.json`（进程内数据，不依赖 vault adapter），跳过 `flushTMVault`：

```ts
// src/app/plugin.ts

// ------ ORIGINAL CODE ------
if (pendingSettings || pendingTranslator) {
    void (async () => {
        if (pendingSettings) await this._saveSettingsImmediate();
        if (pendingTranslator) await this._saveTranslatorDataImmediate();
    })().catch((e) => logger.warn(...));
}
// --------------------------
// ------ NEW CODE ----------
if (pendingSettings || pendingTranslator) {
    void (async () => {
        if (pendingSettings) await this._saveSettingsImmediate();
        // 卸载时序不可靠，跳过依赖 vault adapter 的 flushTMVault，
        // 仅写进程内 data.json（translator 内存数据已含 TM 缓存），保证兜底落盘成功。
        if (pendingTranslator) await this._saveTranslatorDataImmediate(false);
    })().catch((e) => logger.warn("[Chinese Plugin Market] 卸载时落盘失败：", e));
}
// --------------------------
```

并给 `_saveTranslatorDataImmediate` 增加形参 `flushVault = true`，卸载路径传 `false` 跳过 `await this.flushTMVault()`。

**Comment by miaoziguan** [comment]
## 复核：理论成立但实际影响待实证

### 成立部分
`onunload`（`plugin.ts#354-360`）fire-and-forget 调 `_saveTranslatorDataImmediate`，后者 `#563` `await this.flushTMVault()` 访问 `this.app.vault`。

### 不确定点
Obsidian 卸载插件时 `app.vault.adapter` **通常仍可用**（`leaf.detach()` 只拆视图，不卸 vault adapter）。fire-and-forget 的 async 仍在事件循环中执行，是否真的"时序不保证"取决于 Obsidian 内部卸载时序，**无实证**表明此处会失败。

建议：若实测卸载时 TM 笔记确实偶发丢失，则按 issue 修；否则优先级可放最低。
